piff/config.py

```diff
@@ -271,23 +289,27 @@ def meanify(config, logger=None):
     coords = np.concatenate(coords, axis=0)
     logger.info('Computing average for {0} params with {1} stars'.format(len(params[0]), len(coords)))
 
+    if params_fitted is None:
+        params_fitted = range(len(params[0]))
+
     lu_min, lu_max = np.min(coords[:,0]), np.max(coords[:,0])
     lv_min, lv_max = np.min(coords[:,1]), np.max(coords[:,1])
 
     nbin_u = int((lu_max - lu_min) / bin_spacing)
     nbin_v = int((lv_max - lv_min) / bin_spacing)
     binning = [np.linspace(lu_min, lu_max, nbin_u), np.linspace(lv_min, lv_max, nbin_v)]
-    counts, u0, v0 = np.histogram2d(coords[:,0], coords[:,1], bins=binning)
-    counts = counts.T
-    counts = counts.reshape(-1)
-
-    params0 = np.zeros((len(counts),len(params[0])))
+    nbinning = (len(binning[0]) - 1) * (len(binning[1]) - 1)
+    params0 = np.zeros((nbinning, len(params[0])))
 
     for i in range(len(params[0])):
-        average = np.histogram2d(coords[:,0], coords[:,1], bins=binning, weights=params[:,i])[0].T
-        average = average.reshape(-1)
-        average[average!=0] /= counts[average!=0]
-        params0[:,i] = average
+        if i in params_fitted:
+            average, u0, v0, bin_target = binned_statistic_2d(coords[:,0], coords[:,1],
+                                                              params[:,i], bins=binning,
+                                                              statistic=stat_used)
+            average = average.T
+            average = average.reshape(-1)
+            Filter = np.isfinite(average).reshape(-1)
```

Comment:
if I'm understanding this line correctly, then the Filter applied a few lines below will only be for the last param in params_fitted. I don't think Filter is necessarily the same for each parameter. So, I think we want the product of all Filters.

Reply:
Yes indeed, I added that in my last commit